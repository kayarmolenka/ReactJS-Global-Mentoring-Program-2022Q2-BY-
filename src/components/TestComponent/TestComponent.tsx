import React from 'react';

interface TestComponentProps {
  listItems: string[];
}

export const TestComponent = (props: TestComponentProps) => {
  const { listItems } = props;

  return (
    <>
      <div className="test-component">Test Component</div>
      <ul>
        {listItems.map((item) => (
          <li key={item}>{item}</li>
        ))}
      </ul>
    </>
  );
};
