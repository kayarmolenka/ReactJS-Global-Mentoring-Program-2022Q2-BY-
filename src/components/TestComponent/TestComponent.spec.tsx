import { render, screen } from '@testing-library/react';
import React from 'react';
import { TestComponent } from './TestComponent';

describe('TestComponent', () => {
  it('should render item1', () => {
    const { asFragment } = render(<TestComponent listItems={['item1', 'item2']} />);

    expect(screen.getByText('item1')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
