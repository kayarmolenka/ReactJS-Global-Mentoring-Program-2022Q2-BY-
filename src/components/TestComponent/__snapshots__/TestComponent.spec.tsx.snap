// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent should render item1 1`] = `
<DocumentFragment>
  <h1>
    Test
  </h1>
  <div
    class="test-try"
  >
    Test Component
  </div>
  <ul>
    <li>
      item1
    </li>
    <li>
      item2
    </li>
  </ul>
</DocumentFragment>
`;
